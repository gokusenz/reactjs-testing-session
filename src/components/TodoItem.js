import React, { PropTypes } from 'react'
import style from './TodoItem.css'

const TodoItem = ({ text, complete }) => (
  <li className={complete ? style.complete : ''}>{text}</li>
)

TodoItem.prototype = {
  text: PropTypes.string.isRequired,
  complete: PropTypes.bool,
}
export default TodoItem
