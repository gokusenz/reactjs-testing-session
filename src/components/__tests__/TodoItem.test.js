import React from 'react'
import { shallow } from 'enzyme'
import TodoItem from '../TodoItem'

describe('TodoItem', () => {
  let component

  beforeEach(() => {
    component = shallow(<TodoItem text="Hello" />)
  })

  it('matches its snapshot - not complete', () => {
    expect(component).toMatchSnapshot()
  })

  it('matches it snapshot - complete', () => {
    component.setProps({ complete: true })

    expect(component).toMatchSnapshot()
  })

  it('renders correct structure', () => {
    expect(component.is('li')).toBe(true)
  })

  it('has complete class if receiving complete prop', () => {
    component.setProps({ complete: true })

    expect(component.hasClass('complete')).toBe(true)
    expect(component.text()).toBe('Hello :)')
  })
})
