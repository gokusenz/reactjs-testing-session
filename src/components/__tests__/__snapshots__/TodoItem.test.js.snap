// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem matches it snapshot - complete 1`] = `
<li
  className="complete"
>
  Not complete
   :)
</li>
`;

exports[`TodoItem matches its snapshot - not complete 1`] = `
<li
  className="complete"
>
  Note complete
   :)
</li>
`;
